server/src/main/java/org/elasticsearch/index/mapper/TypeFieldMapper.java

```diff
@@ -174,6 +179,36 @@ public Query termsQuery(List<?> values, QueryShardContext context) {
             }
         }
 
+        @Override
+        public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower, boolean includeUpper, QueryShardContext context) {
+            if (hasDocValues()) {
+                return new TermRangeQuery(name(), lowerTerm == null ? null : indexedValueForSearch(lowerTerm),
+                        upperTerm == null ? null : indexedValueForSearch(upperTerm), includeLower, includeUpper);
+            } else {
+                // this means the index has a single type and the type field is implicit
+                DEPRECATION_LOGGER.deprecatedAndMaybeLog("range_single_type",
+                        "Running [range] query on [_type] field for an index with a single type. As types are deprecated, this functionality will be removed in future releases.");
+                Query result = new MatchAllDocsQuery();
+                Collection<String> types = context.getMapperService().types();
```

Comment:
`context.getMapperService().documentMapper()` is missing in 6.x, the existing method requires knowing the type. Should `types()` always return a 1-element collection in the single-type case on 6.x?

Reply:
It may return no types if no mapping has been put, but indeed never more than 1 type. FYI the right way to check is `context.getIndexSettings().isSingleType()` rather than `indexCreatedVersion >= 6.0`